### ISSUE TITLE
Typescript issues following example docs

### ISSUE THREAD

**Issue by jrdn91** [lukemorales/query-key-factory#10]
I'm getting some typescript errors following the examples in the readme and can't quite figure out what they mean or why they are happening.

I copied exactly the `todosKeys` example and am getting the following errors related to specific lines

```
single: (id: string) => id,
```
```
Type '(id: string) => string' is not assignable to type 'FactoryProperty'.
```
```
Type '(id: string) => string' is not assignable to type 'FactoryCallback'.
```
```
Type 'string' is not assignable to type 'AnyObject'.
```
```
Type '(id: string) => string' is not assignable to type 'FactoryProperty'.
  Type '(id: string) => string' is not assignable to type 'FactoryCallback'.
    Type 'string' is not assignable to type 'AnyObject'.ts(2322)
QueryKeys.ts(11, 11): Did you mean to call this expression?
```

One of the other ones is this line

```
search: (query: string, limit: number) => [query, { limit }],
```
```
Type '(query: string, limit: number) => (string | { limit: number; })[]' is not assignable to type 'FactoryProperty'.

```
```
Type '(query: string, limit: number) => (string | { limit: number; })[]' is not assignable to type 'FactoryCallback'.
```
```
Type '(string | { limit: number; })[]' is not assignable to type 'AnyObject'.
```
```
Index signature for type 'string' is missing in type '(string | { limit: number; })[]'.
```
```
(property) search: (query: string, limit: number) => (string | {
    limit: number;
})[]
```
```
Type '(query: string, limit: number) => (string | { limit: number; })[]' is not assignable to type 'FactoryProperty'.
  Type '(query: string, limit: number) => (string | { limit: number; })[]' is not assignable to type 'FactoryCallback'.
    Type '(string | { limit: number; })[]' is not assignable to type 'AnyObject'.
      Index signature for type 'string' is missing in type '(string | { limit: number; })[]'.ts(2322)
```

I love the idea behind your library and would love to figure out why these are happening. I tried upgrading my typescript version to the latest and updating some of the properties in my tsconfig to match the projects but still receive these errors

**Comment by lukemorales** [comment]
Can you share your typescript settings and version so that I can try to reproduce it?

**Comment by jrdn91** [comment]
Yes, the version of typescript that is installed in node_modules is `4.8.2`

Here are the contents of my tsconfig

```

{
  "compilerOptions": {
    /* Basic Options */
    "target": "esnext",                       /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
    "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
    "lib": ["es2017"],                        /* Specify library files to be included in the compilation. */
    "allowJs": true,                          /* Allow javascript files to be compiled. */
    // "checkJs": true,                       /* Report errors in .js files. */
    "jsx": "react-native",                    /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */
    // "declaration": true,                   /* Generates corresponding '.d.ts' file. */
    // "sourceMap": true,                     /* Generates corresponding '.map' file. */
    // "outFile": "./",                       /* Concatenate and emit output to single file. */
    // "outDir": "./",                        /* Redirect output structure to the directory. */
    // "rootDir": "./",                       /* Specify the root directory of input files. Use to control the output directory structure with --outDir. */
    // "removeComments": true,                /* Do not emit comments to output. */
    "noEmit": true,                           /* Do not emit outputs. */
    // "incremental": true,                   /* Enable incremental compilation */
    // "importHelpers": true,                 /* Import emit helpers from 'tslib'. */
    // "downlevelIteration": true,            /* Provide full support for iterables in 'for-of', spread, and destructuring when targeting 'ES5' or 'ES3'. */
    "isolatedModules": true,                  /* Transpile each file as a separate module (similar to 'ts.transpileModule'). */

    /* Strict Type-Checking Options */
    "strict": true,                           /* Enable all strict type-checking options. */
    // "noImplicitAny": true,                 /* Raise error on expressions and declarations with an implied 'any' type. */
    // "strictNullChecks": true,              /* Enable strict null checks. */
    // "strictFunctionTypes": true,           /* Enable strict checking of function types. */
    // "strictPropertyInitialization": true,  /* Enable strict checking of property initialization in classes. */
    // "noImplicitThis": true,                /* Raise error on 'this' expressions with an implied 'any' type. */
    // "alwaysStrict": true,                  /* Parse in strict mode and emit "use strict" for each source file. */

    /* Additional Checks */
    // "noUnusedLocals": true,                /* Report errors on unused locals. */
    // "noUnusedParameters": true,            /* Report errors on unused parameters. */
    // "noImplicitReturns": true,             /* Report error when not all code paths in function return a value. */
    // "noFallthroughCasesInSwitch": true,    /* Report errors for fallthrough cases in switch statement. */

    /* Module Resolution Options */
    "moduleResolution": "node",               /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */
    "baseUrl": ".",                           /* Base directory to resolve non-absolute module names. */
    "paths": {"@/*": ["./src/*"]},             /* A series of entries which re-map imports to lookup locations relative to the 'baseUrl'. */
    // "rootDirs": [],                        /* List of root folders whose combined content represents the structure of the project at runtime. */
    // "typeRoots": [],                       /* List of folders to include type definitions from. */
    // "types": [],                           /* Type declaration files to be included in compilation. */
    "allowSyntheticDefaultImports": true,     /* Allow default imports from modules with no default export. This does not affect code emit, just typechecking. */
    "esModuleInterop": true,                  /* Enables emit interoperability between CommonJS and ES Modules via creation of namespace objects for all imports. Implies 'allowSyntheticDefaultImports'. */
    // "preserveSymlinks": true,              /* Do not resolve the real path of symlinks. */
    "skipLibCheck": true,                     /* Skip type checking of declaration files. */
    "resolveJsonModule": true                 /* Allows importing modules with a ‘.json’ extension, which is a common practice in node projects. */

    /* Source Map Options */
    // "sourceRoot": "./",                    /* Specify the location where debugger should locate TypeScript files instead of source locations. */
    // "mapRoot": "./",                       /* Specify the location where debugger should locate map files instead of generated locations. */
    // "inlineSourceMap": true,               /* Emit a single file with source maps instead of having a separate file. */
    // "inlineSources": true,                 /* Emit the source alongside the sourcemaps within a single file; requires '--inlineSourceMap' or '--sourceMap' to be set. */

    /* Experimental Options */
    // "experimentalDecorators": true,        /* Enables experimental support for ES7 decorators. */
    // "emitDecoratorMetadata": true,         /* Enables experimental support for emitting type metadata for decorators. */
  },
  "exclude": [
    "node_modules", "babel.config.js", "metro.config.js"
  ]
}

```

**Comment by lukemorales** [comment]
Thanks @jrdn91, I'll take a look and try to reproduce it and come back to you later today

**Comment by lukemorales** [comment]
@jrdn91 just figured it out, the new types didn't get published! As soon as #9 is merged, the fix will be released

**Comment by jrdn91** [comment]
> @jrdn91 just figured it out, the new types didn't get published! As soon as #9 is merged, the fix will be released

Good to know I won't have to make a bunch of crazy changes to my typescript setup, sounded stressful :)

**Comment by lukemorales** [comment]
> > @jrdn91 just figured it out, the new types didn't get published! As soon as #9 is merged, the fix will be released
> 
> Good to know I won't have to make a bunch of crazy changes to my typescript setup, sounded stressful :)

The idea is definitely to remove stress rather than adding it to the development lol
It was just a bad compiled version, I'll try to speed the new release

**Comment by jrdn91** [comment]
That fixes the typescript errors, thank you!
